### PR TITLE
feat(form fields): use text presets

### PR DESCRIPTION
**Pull request by anabelengp**
- [x] Input value
- [x] label
- [x] helper
- [x] prefix
- [x] textarea
- [x] suggestions
- [x] select options
- [x] fix long label ellipsis in input with startIcon + endIcon
 image
- [ ] fix long date value ellipsis in datepicker?? ---> I DON'T KNOW HOW  TO SOLVE IT!
image

preint job: https://jenkins.tuenti.io/job/CI-webapp-master-preintegration

### CODE REVIEW

**Comment by anabelengp** [comment]
> please review this: image vs master: image
> 
> All the text field should be clickable not just a small region

done! I broke it in the commit that tried to align prefix + input. I changed the way to align them in https://github.com/Telefonica/mistica-web/pull/359/commits/4c1fd64501be3cd1cb113ed2c93578cb87e07407

**Review by atabel — APPROVED** [review]

**Review by pladaria — APPROVED** [review]

File: flow-defs/text-field-components.js.flow [hunk 1]
```diff
@@ -3,6 +3,9 @@
 import * as React from "react";
```
Comment (yceballost) [inline flow-defs/text-field-components.js.flow:3]: Can we change select arrow asset to mistica icon `chevron-down-regular`? And deprecate the actual one.

image

https://www.figma.com/file/e8d3PkXitUJ1Jwgc9Y3mNc/Forms-Component-Specs?node-id=0%3A1

File: src/text-field-base.tsx [hunk 2]
```diff
@@ -156,23 +165,30 @@ const useStyles = createUseStyles((theme) => ({
...
+    fullWidth: {
+        width: '100%',
+        display: 'inline-flex',
+        alignSelf: ({prefix}) => (prefix ? 'baseline' : 'initial'),
+        '& > div': {
+            width: '100%',
+            display: 'inline-flex',
+        },
     },
     textArea: {
```
Comment (yceballost) [inline src/text-field-base.tsx:177]: TextArea has little margin with desktop presets
image

File: src/text-field-base.tsx [hunk 3]
```diff
@@ -1,8 +1,16 @@
 import * as React from 'react';
```
Comment (yceballost) [inline src/text-field-base.tsx:1]: I was looking that password bullets are smaller than specs
image
https://www.figma.com/file/e8d3PkXitUJ1Jwgc9Y3mNc/Forms-Component-Specs?node-id=0%3A1

Can we solve this in this PR?
We are using in design side, Lucida Grande font in these bullets

**Review by yceballost — APPROVED** [review]
Nice! Everything seems perfect regarding text-presets. I put some comments to change little things. There are things out of text-preset stuff, but would be nice to have it.

I approve this PR and you choose if these changes are for this PR or not!

Thanks!

File: flow-defs/text-field-components.js.flow (hunk 1 above)
Comment (atabel) [reply flow-defs/text-field-components.js.flow:3]: please open a new ticket

File: src/text-field-base.tsx (hunk 3 above)
Comment (atabel) [reply src/text-field-base.tsx:1]: please create a new ticket for this

**Review by HectorSancho — APPROVED** [review]

File: flow-defs/text-field-components.js.flow (hunk 1 above)
Comment (atabel) [reply flow-defs/text-field-components.js.flow:3]: https://github.com/Telefonica/mistica-web/pull/367

**Pull request merged**